pkg/kv/kvserver/stores.go

```diff
@@ -206,12 +207,12 @@ func (ls *Stores) SendWithWriteBytes(
 	return br, writeBytes, pErr
 }
 
-// RangeFeed registers a rangefeed over the specified span. It sends updates to
-// the provided stream and returns with an optional error when the rangefeed is
-// complete.
-func (ls *Stores) RangeFeed(
+// RangeFeedPromise registers a rangefeed over the specified span. It sends
+// updates to the provided stream and returns a future with an optional error
+// when the rangefeed is complete.
+func (ls *Stores) RangeFeedPromise(
```

Comment:
nit: I don't think we really need to rename these methods, they still run rangefeeds. I don't particularly care though.

Reply:
Cool; rename reverted.